thunder/torch/__init__.py

```diff
@@ -2409,6 +2409,29 @@ def tril_(a: TensorLike, /, diagonal: int = 0, *, fill_value: None | Number = No
     return prims.copy_(tril(a, diagonal, fill_value=fill_value), a)
 
 
+# NOTE triu is the same as tril except that we modify the inequality to return the upper triangular
+# NOTE matrix instead of the lower triangular matrix.
+@torchsymbol(torch.triu, is_method=True)
+def triu(a: TensorLike, /, diagonal: int = 0, *, fill_value: None | Number = None) -> TensorLike:
+    utils.check(a.ndim >= 2, lambda: f"triu: a ({a.ndim=}) must have at least two dimensions")
```

Comment:
Alternatively, `triu` is `tril` applied to a transposed input.

Reply:
That sounds good - I'll add that to the comment.

Reply:
But that means that that you can implement `triu` as `tril`. One only has to be careful, because then, if decomposed like that, the result is going to be column-major, and not row-major. But, given that there is a transpose prim, flips from contiguous to column-major should not be a problem? CC @mruberry

Reply:
I'd prefer implement it the way it's done in PyTorch:
https://github.com/pytorch/pytorch/blob/b8f383107eebd9495a0f132d58a970e178e15930/torch/_refs/__init__.py#L5819

Reply:
While it's technically neat to reuse `tril` to implement `triu`, I think the straightforward implementation here that is consistent with PyTorch's decomposition should be OK for now